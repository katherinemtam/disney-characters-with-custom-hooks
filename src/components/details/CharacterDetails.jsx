import React from 'react';
import { useParams } from 'react-router-dom';
import { useCharacter } from '../../state/character';

const CharacterDetails = () => {
  const { id } = useParams();
  const character = useCharacter(id);

  const films = JSON.stringify(character.films);
  const shortFilms = JSON.stringify(character.shortFilms);
  const tvShows = JSON.stringify(character.tvShows);
  const videoGames = JSON.stringify(character.videoGames);
  const parkAttractions = JSON.stringify(character.parkAttractions);
  const allies = JSON.stringify(character.allies);
  const enemies = JSON.stringify(character.enemies);

  return (
    <section>
      <h2>{character.name}</h2>
      <p>Films: {films}</p>
      <p>Short Films: {shortFilms}</p>
      <p>TV Shows: {tvShows}</p>
      <p>Video Games: {videoGames}</p>
      <p>Park Attractions: {parkAttractions}</p>
      <p>Allies: {allies}</p>
      <p>Enemies: {enemies}</p>
    </section>
  );
};

export default CharacterDetails;
