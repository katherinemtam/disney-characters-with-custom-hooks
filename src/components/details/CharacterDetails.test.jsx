import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import CharacterDetails from './CharacterDetails';

describe('CharacterDetails component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetails', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterDetails />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });
});
