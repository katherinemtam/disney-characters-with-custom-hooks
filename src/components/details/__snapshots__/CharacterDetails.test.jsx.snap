// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails component renders CharacterDetails 1`] = `[Function]`;
